core/src/main/java/com/linecorp/armeria/common/encoding/AbstractStreamDecoder.java

```diff
@@ -54,6 +55,10 @@ public HttpData decode(HttpData obj) {
                                               .cause(ex)
                                               .build();
             }
+            throw new UnexpectedDecodeException(
+                    "Unexpected exception has occurred. " +
+                    "This exception is not caused by the buffer reaching max size. Exception message: " +
+                    message);
```

Comment:
I think it is okay to re-throw the exception as is if we don't have additional information for the error message.

That said, your test code is invaluable to verify the code.

Reply:
Your words mean a lot to me. Thank you.